vm-disk-performance-meter/azuredeploy.json

```diff
@@ -0,0 +1,323 @@
+      ],
+      "defaultValue": "1g",
+      "metadata": {
+        "description": "Size of the file to test."
+      }
+    },
+    "secondsToRunTest": {
+      "type": "int",
+      "defaultValue": 30,
+      "metadata": {
+        "description": "Number of seconds for the test to run."
+      }
+    },
+      "adminUsername": {
+        "type": "string",
+        "metadata": {
+          "description": "Username for the test VMs."
+        }
+      },
+      "adminPassword": {
+        "type": "securestring",
+        "metadata": {
+          "description": "Password for the test VMs."
+        }
+      }
+    },
+  "variables": {
+    "osVersion": "14.04.5-LTS",
+    "imagePublisher": "Canonical",
+    "imageOffer": "UbuntuServer",
+    "OSDiskName": "osdiskforwindowssimple",
+    "nicName": "testVMNic",
+    "addressPrefix": "10.0.0.0/16",
+    "subnetName": "Subnet",
+    "subnetPrefix": "10.0.0.0/24",
+    "storageAccountType": "Standard_LRS",
+    "publicIPAddressName": "publicIP",
+    "publicIPAddressType": "Static",
+    "vmStorageAccountContainerName": "vhds",
+    "virtualNetworkName": "testVNET",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
+    "scriptFolder": "scripts",
+    "scriptFileName": "testBandwidth.ps1",
+    "vhdStorageAccountName": "[concat(uniqueString(resourceGroup().id), 'storage')]",
+    "dataStorageAccountName": "[concat(uniqueString(resourceGroup().id), 'datastorage')]",
+    "frontEndNSGName": "[concat('webtestnsg-', uniqueString(resourceGroup().id))]",
+    "vmName": "testVM",
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
+    "qs": "'",
+    "diskCmd": "(echo n; echo p; echo 1; echo; echo; echo w) | fdisk /dev/sdc > /dev/null && mkfs -t ext4 /dev/sdc1 > /dev/null && mkdir /datadisk && mount /dev/sdc1 /datadisk",
+    "installCmd": "apt-get update > /dev/null && apt-get -y install fio > /dev/null",
+    "globalCmd": "[concat('echo [global] > t && echo size=', parameters('testSize'), ' >> t && echo direct=1 >> t && echo iodepth=256 >> t && echo ioengine=libaio >> t && echo bs=8k >> t')]",
+    "workerCmd": "[concat('echo rw=', parameters('testType'), ' >> t && echo directory=/datadisk >> t')]",
+    "cfgCmd": "[concat(variables('globalCmd'), ' && echo [w1] >> t && ', variables('workerCmd'), ' && echo [w2] >> t && ', variables('workerCmd'), ' && echo [w3] >> t && ', variables('workerCmd'), ' && echo [w4] >> t && ', variables('workerCmd'))]",
+    "warmupCmd": "[concat('fio --runtime ', parameters('secondsToRunTest'),  ' t > /dev/null')]",
+    "runCmd": "[concat('fio --runtime ', parameters('secondsToRunTest'),  ' t | grep -E \"READ:|WRITE:\" | tr \"\n\" \";\" | tr -s [:space:] | sed \"s/ :/:/g\" | sed \"s/= /=/g\" ')]",
+    "fullCmd": "[concat('sudo bash -c ', variables('qs'), variables('diskCmd'), ' && ', variables('installCmd'), ' && ', variables('cfgCmd'), ' && ', variables('runCmd'), variables('qs'))]"
```

Comment:
I don't think a human could decipher and/or modify this (though I'm quite impressed you wrote it :)) - can we break it out into an sh file so it's easier to digest and leverage?